src/os/bluestore/KernelDevice.cc

```diff
@@ -375,10 +375,7 @@ int KernelDevice::aio_write(
   assert(off < size);
   assert(off + len <= size);
 
-  if (!bl.is_n_align_sized(block_size) || !bl.is_aligned(block_size)) {
-    dout(20) << __func__ << " rebuilding buffer to be aligned" << dendl;
-    bl.rebuild_aligned(block_size);
-  }
+  bl.rebuild_aligned_size_and_memory(block_size, block_size);
```

Comment:
The only thing we lose here is the debug message saying it had tod o something.  Maybe we can make the bufferlist method return true/false so we know if it had to do a rebuild, and still output the log message?  I'd like to be able to identify cases where a rebuild was needed and ensure the callers can't do a better job aligning their buffers ahead of time.